pkg/util/pki/csr.go

```diff
@@ -109,6 +123,34 @@ func OrganizationForCertificate(crt *v1alpha1.Certificate) []string {
 
 var serialNumberLimit = new(big.Int).Lsh(big.NewInt(1), 128)
 
+func buildUsages(usages []v1alpha1.KeyUsage, isCA bool) (ku x509.KeyUsage, eku []x509.ExtKeyUsage, err error) {
+	var unk []v1alpha1.KeyUsage
+	if isCA {
+		ku |= x509.KeyUsageCertSign
+	}
+	if len(usages) == 0 {
+		usages = append(usages, v1alpha1.DefaultKeyUsages()...)
+	}
+	usages = removeDuplicateKeyUsage(usages)
+	for _, u := range usages {
+		if kuse, ok := apiutil.KeyUsageType(u); ok {
+			ku |= kuse
+		} else if ekuse, ok := apiutil.ExtKeyUsageType(u); ok {
+			eku = append(eku, ekuse)
+		} else {
+			unk = append(unk, u)
+		}
+	}
+	if len(unk) > 0 {
+		err = fmt.Errorf("unknown key usages: %v", unk)
+	}
+	return
+}
+
+func defaultCSRExtKeyUsages() []x509.ExtKeyUsage {
```

Comment:
I think we should set this to empty/remove the function altogether as the `keyUsages` field will be defaulted to the defaults specified in `pkg/apis` once we have actual defaulting support in 0.11.